File: 69337374-main/mario-more/mario.c
#include <stdio.h>
#include <cs50.h>

void pyramid_row(int hashes, int spaces)
{
    for (int j = 0; j < spaces; j++)
    {
        printf(" ");
    }
    for (int j = 0; j < hashes; j++)
    {
        printf("#");
    }
}

void pyramid(int height)
{
    // maximum length, i.e., length of last row
    const int maxlength = height;
    for (int length = 0; length < height; length++)
    {
        // loop over all i from 0 to h-1
        // run `print_pyramid_line` with i + 1
        pyramid_row(length + 1, maxlength - length - 1);
        pyramid_row(length + 1, 2);
        printf("\n");
    }
}

int main(void)
{
    // define height to be 0 currently
    int height = 0;

    // ask for height until the input is correct
    while ((height <= 0) || (height > 8))
    {
        height = get_int("Height: ");
    }

    // call `print_pyramid`
    pyramid(height);
}

File: 69337374-main/sentimental-mario-more/mario.py
from cs50 import get_int


def main():
    height = 0
    # get height from user
    while (height < 1 or height > 8):
        height = get_int("Height: ")

    for bricks in range(1, height + 1):
        # number of spaces to print
        spaces = height - bricks
        print(" " * spaces, end="")    # print spaces
        print("#" * bricks, end="  ")  # print bricks
        print("#" * bricks, end="\n")    # print bricks


if __name__ == "__main__":
    main()
